test/unit/data/symbol_bucket.test.js

```diff
@@ -98,3 +84,25 @@ test('SymbolBucket integer overflow', (t) => {
     t.ok(console.warn.getCall(0).calledWithMatch(/Too many glyphs being rendered in a tile./));
     t.end();
 });
+
+test('SymbolBucket detects rtl text', (t) => {
+    const rtlBucket = bucketSetup('مرحبا');
+    const ltrBucket = bucketSetup('hello');
+    const options = {iconDependencies: {}, glyphDependencies: {}};
+    rtlBucket.populate([{feature}], options);
+    ltrBucket.populate([{feature}], options);
+
+    t.ok(rtlBucket.hasRTLText);
```

Comment:
should we also test `t.ok(!rtlBucket.isEmpty());`? It seems like that's more of what we want to know, since `hasRTLText` is used to determine if the bucket is empty or not.

Reply:
Added a separate test in https://github.com/mapbox/mapbox-gl-js/pull/9091/commits/315742d49d9017da10073566a279664c11c61e63